src/Tmds.Ssh/SshClient.cs

```diff
@@ -676,6 +677,36 @@ private void FreeChannel(uint nr)
             _allocatedChannels[i] = _allocatedChannels[i] & ~mask;
         }
 
+        public async Task<SftpClient> OpenSftpClientAsync(CancellationToken ct)
+        {
+            ChannelContext context = CreateChannel();
+
+            try
+            {
+                // Open the session channel.
+                {
+                    await context.SendChannelOpenSessionMessageAsync(ct).ConfigureAwait(false);
+                    await context.ReceiveChannelOpenConfirmationAsync(ct).ConfigureAwait(false);
+                }
+                // Request command execution.
+                {
+                    await context.SendChannelSubsystemMessageAsync("sftp", ct).ConfigureAwait(false);
+                    await context.ReceiveChannelRequestSuccessAsync("Failed to start sftp.", ct).ConfigureAwait(false);
+                }
+                // {
+                //     await context.SftpInitMessageAsync(3, ct).ConfigureAwait(false);
+                //     // TODO add server negotiation in case server would have min. version < 3, but was able to do a version 3 aswell
+                //     int serverVersion = await context.ReceiveServerVersionAsync("Failed to negotiate SFTP", ct).ConfigureAwait(false);
+                // }
+
+                return new SftpClient(context);
+            }
+            catch
+            {
+                context.Dispose(); // closing channel
```

Comment:
We need to dispose the sftp client in case we've constructed it, similar to:
https://github.com/tmds/Tmds.Ssh/blob/10e93706c323b4ba543da8adb681bc550ad6bd41/src/Tmds.Ssh/SshClient.ExecCommand.cs#L56-L67

Reply:
But only thing the SftpClient would dispose is the context right? Which we already do

Reply:
I guess with the added init there, its true we might do more than channel disposing, maybe of the ReceiveLoopAsync?

Reply:
I think we'll only do Dispose of ChannelContext.
It's a style preference, once we've created the SftpClient, I feel like it own's the ChannelContext so we should Dispose that instead of disposing ChannelContext directly.

Reply:
(After SftpClient was used by the user, Dispose will have more side-effects, like waiting for the ReceiveLoop)

Reply:
https://github.com/tmds/Tmds.Ssh/blob/fa03344365d303ce44dd7976e179343549ad931f/src/Tmds.Ssh/SshClient.Sftp.cs#L21-L33

But because we moved InitAsync there, the dispose should already be able to dispose the ReceiveLoop